Library/Homebrew/os/linux/ld.rb

```diff
@@ -0,0 +1,77 @@
+# typed: true
+# frozen_string_literal: true
+
+module OS
+  module Linux
+    # Helper functions for querying `ld` information.
+    #
+    # @api private
+    module Ld
+      module_function
+
+      sig { returns(String) }
+      def sysconfdir
+        fallback_sysconfdir = "/etc"
+
+        brewed_ld_so = HOMEBREW_PREFIX/"lib/ld.so"
+        return fallback_sysconfdir unless brewed_ld_so.exist?
+
+        ld_so_output = Utils.popen_read(brewed_ld_so, "--list-diagnostics")
+        return fallback_sysconfdir unless $CHILD_STATUS.success?
+
+        match = ld_so_output.match(/path.sysconfdir="(.+)"/)
+        return fallback_sysconfdir unless match
+
+        match.captures.compact.first
+      end
+
+      sig { returns(T::Array[String]) }
+      def system_dirs
+        system_dirs = []
+
+        brewed_ld_so = HOMEBREW_PREFIX/"lib/ld.so"
+        return system_dirs unless brewed_ld_so.exist?
+
+        ld_so_output = Utils.popen_read(brewed_ld_so, "--list-diagnostics").split("\n")
+        return system_dirs unless $CHILD_STATUS.success?
```

Comment:
This is duplicated between the two - maybe worth extracting into another function.